app/lib/core/view_models/account_edit_view_model.dart

```diff
@@ -0,0 +1,54 @@
+import 'package:app/core/models/transporter.dart';
+import 'package:app/core/services/database/database_service.dart';
+import 'package:app/core/services/dialog/dialog_service.dart';
+import 'package:app/core/services/navigation/nav_service.dart';
+import 'package:app/core/services/service_locator.dart';
+import 'package:app/core/view_models/base_view_model.dart';
+import 'package:app/ui/common/view_state.dart';
+import 'package:flutter/material.dart';
+
+class AccountEditViewModel extends BaseViewModel {
+  final DatabaseService _databaseService = locator<DatabaseService>();
+  final NavigationService _navigationService = locator<NavigationService>();
+  final DialogService _dialogService = locator<DialogService>();
+
+  Transporter _account;
+
+  Future<void> saveTransporterAccount({
+    @required String firstName,
+    @required String lastName,
+    @required String userEmailAddress,
+    @required String userPhoneNumber,
+  }) async {
+    setState(ViewState.Busy);
+    var updateTransporterAccount = Transporter(
+      firstName: firstName,
+      isAdmin: _account.isAdmin,
+      lastName: lastName,
+      userEmailAddress: userEmailAddress,
+      userId: _account.userId,
+      userPhoneNumber: userPhoneNumber,
+    );
+    _databaseService.addTransporter(updateTransporterAccount).then((_) =>
```

Comment:
The `addTransporter` function is not an update. This will create duplicates of transporters in the database.

- [ ] Fix by adding `updateTransporter` like the `updateAtr` function in DBS